Improve connector SQL support sections

mosabua opened the pull request:
## Description

* First improve the MERGE limitation and make it a fragment
* Move the limitations into SQL support section, add headers, and improve overview list
* Add for Ignite and Phoenix

## Additional context and related issues

Follow up to #23034 and related PRs. 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

mosabua commented on docs/src/main/sphinx/connector/non-transactional-merge.fragment:
```diff
@@ -0,0 +1,10 @@
+The connector supports adding rows using [MERGE statements](/sql/merge), if the
+`merge.non-transactional-merge.enabled` catalog property or the corresponding
+`non_transactional_merge_enabled` catalog session property is set to 
+`true`. Merge is only supported for directly modifying target tables.
```
Not sure this sentence makes sense and takes the original content and captures what was meant @chenjian2664 and @ebyhr

chenjian2664 reviewed (approved):
Thanks. I was also thought to do this, but forgot :)
Involve @kokosing

chenjian2664 commented on docs/src/main/sphinx/connector/non-transactional-merge.fragment:
```diff
@@ -0,0 +1,10 @@
+### Non-transactional MERGE
+
+The connector supports adding rows using [MERGE statements](/sql/merge), if the
```
`The connector supports adding rows using` MERGE can add/update/delete rows, maybe you have better phrase? like change `adding rows` to `modifying rows`?

mosabua replied:
Updated

chenjian2664 commented:
Phoenix Connector also supports non transactional merge @mosabua

mosabua commented:
Ignite connector also needs to be updated

mosabua commented:
I think this is good to go now .. Updated all JDBC connectors and made them more consistent. Also added the Ignite and Pheonx merge .. all in clean separate commits .. I think we should work towards merge and I can revisit the whole stuff more for other connectors.

mosabua commented:
Ping @ebyhr or @electrum

electrum commented on docs/src/main/sphinx/connector/non-transactional-merge.fragment:
```diff
@@ -0,0 +1,11 @@
+### Non-transactional MERGE
+
+The connector supports adding, updatingm, and deleting rows using [MERGE
```
Typo "updating"

electrum commented on docs/src/main/sphinx/connector/non-transactional-merge.fragment:
```diff
@@ -0,0 +1,11 @@
+property is set to `true`. Merge is only supported for directly modifying target
+tables.
+
+Note that in rare cases expections occur during the merge operation, and no
```
> In rare cases, errors may occur during the merge operation, potentially resulting in a partial update.
